LocalFolder/explainer.md

```diff
@@ -0,0 +1,129 @@
+# ApplicationData.LocalFolder Access From Microsoft Store PWA 
+
+Author: [Lu Huang](https://github.com/luhuangMSFT)
+
+## Status of this Document
+
+This document is a starting point for engaging the community and standards bodies in developing collaborative solutions fit for standardization. As the solutions to problems described in this document progress along the standards-track, we will retain this document as an archive and use this section to keep the community up-to-date with the most current standards venue and content location of future work and discussions.
+
+* This document status: **Active**
+* Expected venue: [Microsoft Edge Explainers](https://github.com/MicrosoftEdge/MSEdgeExplainers)
+* **Current version: this document**
+
+## Introduction
+[UWP apps](https://learn.microsoft.com/en-us/windows/uwp/get-started/universal-application-platform-guide) often make use of the [`ApplicationData.LocalFolder`](https://learn.microsoft.com/en-us/uwp/api/windows.storage.applicationdata.localfolder) API for access to local storage. This API manages storage in a system file directory unique to the Package Family Name (PFN) of the installed app package. A UWP app distributed by the Microsoft Store may be replaced with a PWA app under the same PFN. When this happens, existing files in the `LocalFolder` directory become inaccessible to the PWA. 
+
+In this explainer, we propose a solution that allows Microsoft Store PWAs to read and delete files from the `LocalFolder` directory belonging to their PFN. This allows apps to provide a more seamless user experience after an update as well as reclaim storage space.
+
+The proposed solution works by making use of the [Origin Private File System (OPFS)](https://web.dev/articles/origin-private-file-system) and the [File System Access API](https://wicg.github.io/file-system-access/). It exposes an app's LocalFolder file system directory as an entry in the app origin's OPFS root directory.
+
+## Goals
+
+* Enable a Microsoft Store PWA to:
+   * read files from the `LocalFolder` directory specific to its PFN.
+   * delete any file and sub-directory in its `LocalFolder` directory. 
+* Avoid prompting end users to provide input or confirmation.
+* Avoid naming collisions with other OPFS root entries. 
+
+## Non-goals
+
+* Enable a Microsoft Store PWA to create new files or modify existing files in its LocalFolder directory.
+* Support web apps not installed and managed by the Microsoft Store.
+* Support web apps on non-Windows platforms.
+* Standardization as a web specification.
+* Introduce a non-standard Web API
+
+## Proposed Solution 
+
+### Existing behavior
+
+`navigator.storage.getDirectory()` is an existing API that returns a promise to a FileSystemDirectoryHandle representing the root of the origin's OPFS storage space. By default, an origin's OPFS root directory will have no entries.
```

Comment:
nit: Would be helpful to link to have these existing API's directly link to their corresponding documentation.

```suggestion
[`navigator.storage.getDirectory()`](https://developer.mozilla.org/en-US/docs/Web/API/StorageManager/getDirectory) is an existing API that returns a promise to a [`FileSystemDirectoryHandle`] (https://developer.mozilla.org/en-US/docs/Web/API/FileSystemDirectoryHandle) representing the root of the origin's OPFS storage space. By default, an origin's OPFS root directory will have no entries.
```